TestRequestRemoteRenameChanged is racy/flaky

Noticed a failure on CI recently and I get a failure locally too:

```
$ for i in {1..100}; do echo $i/100; go test -race -count 1 -run RemoteRenameChanged ./lib/model/ || break; done
1/100
ok      github.com/syncthing/syncthing/lib/model        0.568s
2/100
ok      github.com/syncthing/syncthing/lib/model        0.573s
3/100
ok      github.com/syncthing/syncthing/lib/model        0.569s
[...]
36/100
ok      github.com/syncthing/syncthing/lib/model        0.570s
37/100
ok      github.com/syncthing/syncthing/lib/model        0.583s
38/100
15:47:13 INFO: No stored folder metadata for "default"; recalculating
15:47:13 INFO: Ready to synchronize "default" (default) (sendreceive)
15:47:13 INFO: Completed initial scan of sendreceive folder "default" (default)
15:47:13 INFO: Device AIR6LPZ-7K4PTTV-UXQSMUU-CPQ5YWH-OEDFIIQ-JUG777G-2YQXXR5-YD6AWQR client is " " named "" at 
15:47:14 INFO: Puller (folder "default" (default), item "b"): handling file: file modified but not rescanned; will try again later
15:47:14 INFO: "default" (default): Failed to sync 1 items
15:47:14 INFO: Folder "default" (default) isn't making sync progress - retrying in 1m0.017223322s.
15:47:24 INFO: Connection to AIR6LPZ-7K4PTTV-UXQSMUU-CPQ5YWH-OEDFIIQ-JUG777G-2YQXXR5-YD6AWQR at  closed: Syncthing is being stopped
--- FAIL: TestRequestRemoteRenameChanged (10.04s)
    requests_test.go:808: timed out without receiving all expected index updates
FAIL
FAIL    github.com/syncthing/syncthing/lib/model        10.065s
FAIL
```

--- a/lib/model/folder_sendrecv.go
+++ b/lib/model/folder_sendrecv.go
@@ -471,7 +471,7 @@ nextFile:
 			// map.
 			desired := fileDeletions[candidate.Name]
 			if err := f.renameFile(candidate, desired, fi, snap, dbUpdateChan, scanChan); err != nil {
-				l.Debugln("rename shortcut for %s failed: %S", fi.Name, err.Error())
+				l.Debugf("rename shortcut for %s failed: %S", fi.Name, err.Error())
 				// Failed to rename, try next one.
 				continue
 			}

--- a/lib/model/folder_sendrecv.go
+++ b/lib/model/folder_sendrecv.go
@@ -471,7 +471,7 @@ nextFile:
 			// map.
 			desired := fileDeletions[candidate.Name]
 			if err := f.renameFile(candidate, desired, fi, snap, dbUpdateChan, scanChan); err != nil {
-				l.Debugf("rename shortcut for %s failed: %S", fi.Name, err.Error())
+				l.Debugf("rename shortcut for %s failed: %s", fi.Name, err.Error())
 				// Failed to rename, try next one.
 				continue
 			}

--- a/lib/model/requests_test.go
+++ b/lib/model/requests_test.go
@@ -720,8 +720,6 @@ func TestRequestRemoteRenameChanged(t *testing.T) {
 	}
 
 	var gotA, gotB, gotConfl bool
-	bIntermediateVersion := protocol.Vector{}.Update(fc.id.Short()).Update(myID.Short())
-	bFinalVersion := bIntermediateVersion.Copy().Update(fc.id.Short())
 	done := make(chan struct{})
 	fc.mut.Lock()
 	fc.indexFn = func(_ context.Context, folder string, fs []protocol.FileInfo) {
@@ -742,16 +740,12 @@ func TestRequestRemoteRenameChanged(t *testing.T) {
 				if gotB {
 					t.Error("Got more than one index update for", f.Name)
 				}
-				if f.Version.Equal(bIntermediateVersion) {
+				if f.Version.Counter(fc.id.Short()) == 0 {
 					// This index entry might be superseeded
 					// by the final one or sent before it separately.
 					break
 				}
-				if f.Version.Equal(bFinalVersion) {
-					gotB = true
-					break
-				}
-				t.Errorf("Got unexpected version %v for file %v in index update", f.Version, f.Name)
+				gotB = true
 			case strings.HasPrefix(f.Name, "b.sync-conflict-"):
 				if gotConfl {
 					t.Error("Got more than one index update for conflicts of", f.Name)

